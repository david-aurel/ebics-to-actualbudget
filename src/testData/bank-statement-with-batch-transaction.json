{
  "Document": {
    "$": { "xmlns": "urn:iso:std:iso:20022:tech:xsd:camt.053.001.04" },
    "BkToCstmrStmt": [
      {
        "GrpHdr": [
          {
            "MsgId": ["E00000002E244711-00001"],
            "CreDtTm": ["2001-02-02T12:30:00.000+02:00"],
            "MsgPgntn": [{ "PgNb": ["1"], "LastPgInd": ["true"] }],
            "AddtlInf": ["SPS/1.7/PROD"]
          }
        ],
        "Stmt": [
          {
            "Id": ["xxx"],
            "ElctrncSeqNb": ["82"],
            "CreDtTm": ["2001-02-02T00:29:29.532+02:00"],
            "FrToDt": [
              {
                "FrDtTm": ["2001-02-02T00:00:00.000000000+02:00"],
                "ToDtTm": ["2001-02-02T23:59:59.999999999+02:00"]
              }
            ],
            "Acct": [
              {
                "Id": [{ "IBAN": ["xxx"] }],
                "Ownr": [
                  {
                    "Nm": ["Spongebob Squarepants"],
                    "PstlAdr": [
                      {
                        "StrtNm": ["Conch St."],
                        "BldgNb": ["124"],
                        "PstCd": [""],
                        "TwnNm": ["Bikini Bottom"],
                        "Ctry": ["CH"]
                      }
                    ]
                  }
                ],
                "Svcr": [
                  {
                    "FinInstnId": [
                      {
                        "BICFI": ["xxx"],
                        "Othr": [{ "Id": ["xxx"], "Issr": ["VAT-ID"] }]
                      }
                    ]
                  }
                ]
              }
            ],
            "Bal": [
              {
                "Tp": [{ "CdOrPrtry": [{ "Cd": ["OPBD"] }] }],
                "Amt": [{ "_": "xxx", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "Dt": [{ "Dt": ["2001-02-02"] }]
              },
              {
                "Tp": [{ "CdOrPrtry": [{ "Cd": ["CLBD"] }] }],
                "Amt": [{ "_": "xxx", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "Dt": [{ "Dt": ["2001-02-02"] }]
              },
              {
                "Tp": [{ "CdOrPrtry": [{ "Cd": ["CLAV"] }] }],
                "Amt": [{ "_": "xxx", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "Dt": [{ "Dt": ["2001-02-02"] }]
              }
            ],
            "TxsSummry": [
              {
                "TtlNtries": [
                  {
                    "NbOfNtries": ["8"],
                    "TtlNetNtry": [{ "Amt": ["xxxx"], "CdtDbtInd": ["DBIT"] }]
                  }
                ],
                "TtlCdtNtries": [{ "NbOfNtries": ["3"], "Sum": ["xxx"] }],
                "TtlDbtNtries": [{ "NbOfNtries": ["5"], "Sum": ["xxx"] }]
              }
            ],
            "Ntry": [
              {
                "Amt": [{ "_": "43.25", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-02-01"] }],
                "AcctSvcrRef": ["L113R111895HKC4B-2"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["RCDT"], "SubFmlyCd": ["AUTT"] }]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": [
                  "Gutschrift TWINT: Ut Dolor Dapibus Industries"
                ]
              },
              {
                "Amt": [{ "_": "167.5", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-02-02"] }],
                "AcctSvcrRef": ["L113R111899M5L78-2"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["RCDT"], "SubFmlyCd": ["AUTT"] }]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": [
                  "Gutschrift TWINT: Ut Dolor Dapibus Industries"
                ]
              },
              {
                "Amt": [{ "_": "100", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-02-01"] }],
                "AcctSvcrRef": ["L113P111899MFNNS-2"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["RCDT"], "SubFmlyCd": ["AUTT"] }]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": ["Gutschrift TWINT: Nunc In Consulting"]
              },
              {
                "Amt": [{ "_": "15", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["DBIT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-02-01"] }],
                "AcctSvcrRef": ["L113P111899WMS2X-1"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["ICDT"], "SubFmlyCd": ["AUTT"] }]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": ["Belastung TWINT: , Arcu LL"]
              },
              {
                "Amt": [{ "_": "3086.5", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["DBIT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-01-30"] }],
                "AcctSvcrRef": ["Z241201318625"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["ICDT"], "SubFmlyCd": ["AUTT"] }]
                      }
                    ]
                  }
                ],
                "Chrgs": [""],
                "NtryDtls": [
                  {
                    "Btch": [
                      {
                        "NbOfTxs": ["2"],
                        "TtlAmt": [{ "_": "3086.5", "$": { "Ccy": "CHF" } }]
                      }
                    ],
                    "TxDtls": [
                      {
                        "Refs": [{ "AcctSvcrRef": ["MO2404264911A8DD"] }],
                        "Amt": [{ "_": "86.5", "$": { "Ccy": "CHF" } }],
                        "CdtDbtInd": ["DBIT"],
                        "BkTxCd": [
                          {
                            "Domn": [
                              {
                                "Cd": ["PMNT"],
                                "Fmly": [
                                  { "Cd": ["ICDT"], "SubFmlyCd": ["DMCT"] }
                                ]
                              }
                            ]
                          }
                        ],
                        "RltdPties": [
                          {
                            "Dbtr": [
                              {
                                "Nm": ["Crusty Crab AG"],
                                "PstlAdr": [
                                  {
                                    "Ctry": ["LI"],
                                    "AdrLine": [
                                      "Crusty Crab AG",
                                      "Bikini Bottom"
                                    ]
                                  }
                                ]
                              }
                            ],
                            "Cdtr": [
                              {
                                "Nm": ["Spongebob Squarepants"],
                                "PstlAdr": [
                                  {
                                    "StrtNm": ["Conch St."],
                                    "BldgNb": ["124"],
                                    "PstCd": [""],
                                    "TwnNm": ["Bikini Bottom"],
                                    "Ctry": ["CH"]
                                  }
                                ]
                              }
                            ],
                            "CdtrAcct": [{ "Id": [{ "IBAN": ["xxx"] }] }]
                          }
                        ],
                        "RltdAgts": [
                          {
                            "DbtrAgt": [
                              { "FinInstnId": [{ "BICFI": ["xxx"] }] }
                            ],
                            "CdtrAgt": [
                              {
                                "FinInstnId": [
                                  {
                                    "ClrSysMmbId": [{ "MmbId": ["xxx"] }],
                                    "PstlAdr": [
                                      {
                                        "AdrLine": ["Bank", "Postfach", "9999"]
                                      }
                                    ]
                                  }
                                ]
                              }
                            ]
                          }
                        ],
                        "RmtInf": [
                          {
                            "Strd": [
                              {
                                "CdtrRefInf": [
                                  {
                                    "Tp": [
                                      { "CdOrPrtry": [{ "Prtry": ["QRR"] }] }
                                    ],
                                    "Ref": ["000000161897617040000275204"]
                                  }
                                ]
                              }
                            ]
                          }
                        ]
                      },
                      {
                        "Refs": [{ "AcctSvcrRef": ["MO2404284914D1B9"] }],
                        "Amt": [{ "_": "3000", "$": { "Ccy": "CHF" } }],
                        "CdtDbtInd": ["DBIT"],
                        "BkTxCd": [
                          {
                            "Domn": [
                              {
                                "Cd": ["PMNT"],
                                "Fmly": [
                                  { "Cd": ["ICDT"], "SubFmlyCd": ["DMCT"] }
                                ]
                              }
                            ]
                          }
                        ],
                        "RltdPties": [
                          {
                            "Dbtr": [
                              {
                                "Nm": ["Spongebob Squarepants"],
                                "PstlAdr": [
                                  {
                                    "AdrLine": [
                                      "Spongebob Squarepants",
                                      "Conch St. 124",
                                      "Bikini Bottom",
                                      "CH"
                                    ]
                                  }
                                ]
                              }
                            ],
                            "Cdtr": [
                              {
                                "Nm": ["Spongebob Squarepants"],
                                "PstlAdr": [{ "Ctry": ["CH"] }]
                              }
                            ],
                            "CdtrAcct": [{ "Id": [{ "IBAN": ["xxx"] }] }]
                          }
                        ],
                        "RltdAgts": [
                          {
                            "DbtrAgt": [
                              { "FinInstnId": [{ "BICFI": ["xxx"] }] }
                            ],
                            "CdtrAgt": [
                              {
                                "FinInstnId": [
                                  {
                                    "ClrSysMmbId": [{ "MmbId": ["xxx"] }],
                                    "PstlAdr": [
                                      {
                                        "AdrLine": ["Bank", "Postfach", "9999"]
                                      }
                                    ]
                                  }
                                ]
                              }
                            ]
                          }
                        ]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": [
                  "Belastungen eBanking Mobile (2): Auftrags-Nr. xxx"
                ]
              },
              {
                "Amt": [{ "_": "0.59", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["DBIT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-02-01"] }],
                "AcctSvcrRef": ["L115B11189B342GP"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["CCRD"], "SubFmlyCd": ["POSD"] }]
                      }
                    ]
                  }
                ],
                "CardTx": [
                  {
                    "Card": [
                      {
                        "PlainCardData": [
                          {
                            "PAN": ["xxx"],
                            "CardSeqNb": ["xxx"],
                            "XpryDt": ["xxx"]
                          }
                        ],
                        "CardBrnd": [{ "Id": ["50"] }]
                      }
                    ],
                    "POI": [
                      { "Id": [{ "Id": ["EasyPark GmbH 00000 easypark.de"] }] }
                    ]
                  }
                ],
                "AddtlNtryInf": [
                  "Online-Einkauf aus Abonnement ZKB Visa Debit Card Nr. xxxx, EasyPark GmbH 00000"
                ]
              },
              {
                "Amt": [{ "_": "51.45", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["DBIT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-02-01"] }],
                "AcctSvcrRef": ["L115B11189B38KX5"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["CCRD"], "SubFmlyCd": ["POSD"] }]
                      }
                    ]
                  }
                ],
                "CardTx": [
                  {
                    "Card": [
                      {
                        "PlainCardData": [
                          {
                            "PAN": ["xxx"],
                            "CardSeqNb": ["xxx"],
                            "XpryDt": ["xxx"]
                          }
                        ],
                        "CardBrnd": [{ "Id": ["50"] }]
                      }
                    ],
                    "POI": [{ "Id": [{ "Id": ["Migros M"] }] }]
                  }
                ],
                "AddtlNtryInf": [
                  "Online-Einkauf ZKB Visa Debit Card Nr. xxxx, Migros"
                ]
              },
              {
                "Amt": [{ "_": "5.6", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["DBIT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-02"] }],
                "ValDt": [{ "Dt": ["2001-02-01"] }],
                "AcctSvcrRef": ["L115B11189B4Q78V"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["CCRD"], "SubFmlyCd": ["POSD"] }]
                      }
                    ]
                  }
                ],
                "CardTx": [
                  {
                    "Card": [
                      {
                        "PlainCardData": [
                          {
                            "PAN": ["xxx"],
                            "CardSeqNb": ["xxx"],
                            "XpryDt": ["xxx"]
                          }
                        ],
                        "CardBrnd": [{ "Id": ["50"] }]
                      }
                    ],
                    "POI": [{ "Id": [{ "Id": ["Migros M"] }] }]
                  }
                ],
                "AddtlNtryInf": [
                  "Online-Einkauf ZKB Visa Debit Card Nr. xxxx, Migros M"
                ]
              }
            ]
          }
        ]
      }
    ]
  }
}
