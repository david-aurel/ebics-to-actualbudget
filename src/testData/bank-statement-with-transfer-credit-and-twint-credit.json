{
  "Document": {
    "$": { "xmlns": "urn:iso:std:iso:20022:tech:xsd:camt.053.001.04" },
    "BkToCstmrStmt": [
      {
        "GrpHdr": [
          {
            "MsgId": ["xxx"],
            "CreDtTm": ["2001-02-04T12:30:00.000+02:00"],
            "MsgPgntn": [{ "PgNb": ["1"], "LastPgInd": ["true"] }],
            "AddtlInf": ["SPS/1.7/PROD"]
          }
        ],
        "Stmt": [
          {
            "Id": ["xxx"],
            "ElctrncSeqNb": ["7"],
            "CreDtTm": ["2001-02-04T12:30:00.000+02:00"],
            "FrToDt": [
              {
                "FrDtTm": ["2001-02-04T00:00:00.000000000+02:00"],
                "ToDtTm": ["2001-02-04T23:59:59.999999999+02:00"]
              }
            ],
            "Acct": [
              {
                "Id": [{ "IBAN": ["xxx"] }],
                "Ownr": [
                  {
                    "Nm": ["Spongebob Squarepants"],
                    "PstlAdr": [
                      {
                        "StrtNm": ["Conch St."],
                        "BldgNb": ["124"],
                        "PstCd": [""],
                        "TwnNm": ["Bikini Bottom"],
                        "Ctry": ["CH"]
                      }
                    ]
                  }
                ],
                "Svcr": [
                  {
                    "FinInstnId": [
                      {
                        "BICFI": ["xxx"],
                        "Othr": [{ "Id": ["xxx"], "Issr": ["VAT-ID"] }]
                      }
                    ]
                  }
                ]
              }
            ],
            "Bal": [
              {
                "Tp": [{ "CdOrPrtry": [{ "Cd": ["OPBD"] }] }],
                "Amt": [{ "_": "xxx", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "Dt": [{ "Dt": ["2001-02-04"] }]
              },
              {
                "Tp": [{ "CdOrPrtry": [{ "Cd": ["CLBD"] }] }],
                "Amt": [{ "_": "xxx", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "Dt": [{ "Dt": ["2001-02-04"] }]
              },
              {
                "Tp": [{ "CdOrPrtry": [{ "Cd": ["CLAV"] }] }],
                "Amt": [{ "_": "xxx", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "Dt": [{ "Dt": ["2001-02-04"] }]
              }
            ],
            "TxsSummry": [
              {
                "TtlNtries": [
                  {
                    "NbOfNtries": ["3"],
                    "TtlNetNtry": [{ "Amt": ["xxx"], "CdtDbtInd": ["CRDT"] }]
                  }
                ],
                "TtlCdtNtries": [{ "NbOfNtries": ["2"], "Sum": ["xxx"] }],
                "TtlDbtNtries": [{ "NbOfNtries": ["1"], "Sum": ["xxx"] }]
              }
            ],
            "Ntry": [
              {
                "NtryRef": ["xxx"],
                "Amt": [{ "_": "10", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-04"] }],
                "ValDt": [{ "Dt": ["2001-02-02"] }],
                "AcctSvcrRef": ["YEP66MP33MMUH"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["RCDT"], "SubFmlyCd": ["DMCT"] }]
                      }
                    ]
                  }
                ],
                "Chrgs": [""],
                "NtryDtls": [
                  {
                    "TxDtls": [
                      {
                        "Refs": [
                          {
                            "AcctSvcrRef": ["FBEHKK9UYYKZNFUH"],
                            "PmtInfId": ["xxx"]
                          }
                        ],
                        "Amt": [{ "_": "10", "$": { "Ccy": "CHF" } }],
                        "CdtDbtInd": ["CRDT"],
                        "BkTxCd": [
                          {
                            "Domn": [
                              {
                                "Cd": ["PMNT"],
                                "Fmly": [
                                  { "Cd": ["RCDT"], "SubFmlyCd": ["DMCT"] }
                                ]
                              }
                            ]
                          }
                        ],
                        "RltdPties": [
                          {
                            "Dbtr": [
                              {
                                "Nm": ["Crusty Crab AG"],
                                "PstlAdr": [
                                  {
                                    "Ctry": ["CH"],
                                    "AdrLine": ["Crusty Crab", "Bikini Bottom"]
                                  }
                                ]
                              }
                            ]
                          }
                        ],
                        "RltdAgts": [
                          {
                            "CdtrAgt": [
                              { "FinInstnId": [{ "BICFI": ["xxx"] }] }
                            ]
                          }
                        ],
                        "RmtInf": [{ "Ustrd": ["Patty Reimbursement"] }]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": ["Gutschrift: Auftrags-Nr. xxx"]
              },
              {
                "Amt": [{ "_": "258.5", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["DBIT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-04"] }],
                "ValDt": [{ "Dt": ["2001-02-04"] }],
                "AcctSvcrRef": ["C4CBEGNHGCABB"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["ICDT"], "SubFmlyCd": ["AUTT"] }]
                      }
                    ]
                  }
                ],
                "Chrgs": [""],
                "NtryDtls": [
                  {
                    "TxDtls": [
                      {
                        "Refs": [{ "AcctSvcrRef": ["EN8LPW4HLNMQVKLM"] }],
                        "Amt": [{ "_": "258.5", "$": { "Ccy": "CHF" } }],
                        "CdtDbtInd": ["DBIT"],
                        "BkTxCd": [
                          {
                            "Domn": [
                              {
                                "Cd": ["PMNT"],
                                "Fmly": [
                                  { "Cd": ["ICDT"], "SubFmlyCd": ["DMCT"] }
                                ]
                              }
                            ]
                          }
                        ],
                        "RltdPties": [
                          {
                            "Dbtr": [
                              {
                                "PstlAdr": [
                                  {
                                    "AdrLine": [
                                      "Spongebob Squarepants",
                                      "Conch St. 124",
                                      "Bikini Bottom",
                                      "CH"
                                    ]
                                  }
                                ]
                              }
                            ],
                            "Cdtr": [
                              {
                                "Nm": ["Tempus Scelerisque Incorporated"],
                                "PstlAdr": [""]
                              }
                            ],
                            "CdtrAcct": [{ "Id": [{ "IBAN": ["xxx"] }] }]
                          }
                        ],
                        "RltdAgts": [
                          {
                            "DbtrAgt": [
                              { "FinInstnId": [{ "BICFI": ["xxx"] }] }
                            ],
                            "CdtrAgt": [
                              {
                                "FinInstnId": [
                                  {
                                    "ClrSysMmbId": [{ "MmbId": ["xxx"] }],
                                    "PstlAdr": [
                                      {
                                        "AdrLine": [
                                          "Bikini Bottom Bank",
                                          "Postfach",
                                          "Bikini Bottom"
                                        ]
                                      }
                                    ]
                                  }
                                ]
                              }
                            ]
                          }
                        ],
                        "RmtInf": [
                          {
                            "Strd": [
                              {
                                "CdtrRefInf": [
                                  {
                                    "Tp": [
                                      { "CdOrPrtry": [{ "Prtry": ["QRR"] }] }
                                    ],
                                    "Ref": [
                                      "Montly Payment for Pineapple Insurance"
                                    ]
                                  }
                                ]
                              }
                            ]
                          }
                        ]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": ["Belastung eBanking Mobile: Auftrags-Nr. xxx"]
              },
              {
                "Amt": [{ "_": "129.25", "$": { "Ccy": "CHF" } }],
                "CdtDbtInd": ["CRDT"],
                "RvslInd": ["false"],
                "Sts": ["BOOK"],
                "BookgDt": [{ "Dt": ["2001-02-04"] }],
                "ValDt": [{ "Dt": ["2001-02-04"] }],
                "AcctSvcrRef": ["PJ3HJVPJJBKRXBN6-2"],
                "BkTxCd": [
                  {
                    "Domn": [
                      {
                        "Cd": ["PMNT"],
                        "Fmly": [{ "Cd": ["RCDT"], "SubFmlyCd": ["AUTT"] }]
                      }
                    ]
                  }
                ],
                "AddtlNtryInf": ["Gutschrift TWINT: Star, Patrick xxx"]
              }
            ]
          }
        ]
      }
    ]
  }
}
