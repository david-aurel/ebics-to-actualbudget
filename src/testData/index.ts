import bank_statement_with_card_debit from './bank-statement-with-card-debit.json'
import bank_statement_with_standing_order_and_ebill from './bank-statement-with-standing-order-and-ebill.json'
import bank_statement_with_transfer_credit_and_twint_credit from './bank-statement-with-transfer-credit-and-twint-credit.json'
import bank_statement_with_transfer_debit from './bank-statement-with-transfer-debit.json'
import bank_statement_with_twint_debit from './bank-statement-with-twint-debit.json'
import bank_statement_without_entries from './bank-statement-without-entries.json'

export const testData = [
  bank_statement_with_card_debit,
  bank_statement_with_standing_order_and_ebill,
  bank_statement_with_transfer_credit_and_twint_credit,
  bank_statement_with_transfer_debit,
  bank_statement_with_twint_debit,
  bank_statement_without_entries,
]
