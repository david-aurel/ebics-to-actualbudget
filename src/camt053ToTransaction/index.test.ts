import { camt053ToTransaction } from '.'
import { testData } from '../testData'

test('The `camt053ToTransaction` function converts camt053 JSON into actual-budget transaction objects', () => {
  const results = testData.flatMap((data) =>
    camt053ToTransaction(JSON.stringify(data))
  )
  expect(results).toMatchSnapshot()
})
