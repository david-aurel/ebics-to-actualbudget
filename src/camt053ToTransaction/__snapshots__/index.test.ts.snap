// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The \`camt053ToTransaction\` function converts camt053 JSON into actual-budget transaction objects 1`] = `
[
  {
    "account": "test-account-id",
    "amount": 5000,
    "cleared": true,
    "date": "2001-02-02",
    "id": "RE22RVOIVYJBCYZK-1",
    "imported_id": "RE22RVOIVYJBCYZK-1",
    "imported_payee": "TWINT: , Ut Dolor Dapibus Industries",
    "notes": "Belastung TWINT: , Ut Dolor Dapibus Industries",
    "payee_name": "TWINT: , Ut Dolor Dapibus Industries",
  },
  {
    "account": "test-account-id",
    "amount": 230,
    "cleared": true,
    "date": "2001-02-02",
    "id": "3UQQCECHVU24Q66L",
    "imported_id": "3UQQCECHVU24Q66L",
    "imported_payee": "Nunc In Consulting",
    "notes": "Online-Einkauf ZKB Visa Debit Card Nr. xxxx 0xxx, Nunc In Consulting",
    "payee_name": "Nunc In Consulting",
  },
  {
    "account": "test-account-id",
    "amount": 230,
    "cleared": true,
    "date": "2001-02-02",
    "id": "VP7VDRG7LCXUROLX",
    "imported_id": "VP7VDRG7LCXUROLX",
    "imported_payee": "Arcu LL",
    "notes": "Online-Einkauf ZKB Visa Debit Card Nr. xxxx 0xxx, Arcu LL",
    "payee_name": "Arcu LL",
  },
  {
    "account": "test-account-id",
    "amount": 2480,
    "cleared": true,
    "date": "2001-02-02",
    "id": "YNBA2JZDRTUPHVVY",
    "imported_id": "YNBA2JZDRTUPHVVY",
    "imported_payee": "Crusty Crab Restaurant Bikini Bottom",
    "notes": "Einkauf ZKB Visa Debit Card Nr. xxxx 0xxx, Crusty Crab Restaurant Bikini Bottom",
    "payee_name": "Crusty Crab Restaurant Bikini Bottom",
  },
  {
    "account": "test-account-id",
    "amount": 150,
    "cleared": true,
    "date": "2001-02-02",
    "id": "TZCCJDWO7M7WZHDP",
    "imported_id": "TZCCJDWO7M7WZHDP",
    "imported_payee": "Crusty Crab Restaurant Bikini Bottom",
    "notes": "Einkauf ZKB Visa Debit Card Nr. xxxx 0xxx, Crusty Crab Restaurant Bikini Bottom",
    "payee_name": "Crusty Crab Restaurant Bikini Bottom",
  },
  {
    "account": "test-account-id",
    "amount": -999925,
    "cleared": true,
    "date": "2001-02-02",
    "id": "Z7AFWZ4V6XCEZ",
    "imported_id": "Z7AFWZ4V6XCEZ",
    "imported_payee": "Crusty Crab AG",
    "notes": "Gutschrift: Auftrags-Nr. xxx (Salaerzahlung)",
    "payee_name": "Crusty Crab AG",
  },
  {
    "account": "test-account-id",
    "amount": 28200,
    "cleared": true,
    "date": "2001-02-05",
    "id": "YZUCQITEOSPXC",
    "imported_id": "YZUCQITEOSPXC",
    "imported_payee": "Fringilla Cursus Purus LLC",
    "notes": "Belastung eBill: Auftrags-Nr. xxx (123456789)",
    "payee_name": "Fringilla Cursus Purus LLC",
  },
  {
    "account": "test-account-id",
    "amount": 58800,
    "cleared": true,
    "date": "2001-02-05",
    "id": "YFE6VNWZATTKB",
    "imported_id": "YFE6VNWZATTKB",
    "imported_payee": "Eu Dolor Foundation",
    "notes": "Belastung Dauerauftrag: Auftrags-Nr. xxx (Monthly Payment for Pineapple Insurance)",
    "payee_name": "Eu Dolor Foundation",
  },
  {
    "account": "test-account-id",
    "amount": 230,
    "cleared": true,
    "date": "2001-02-05",
    "id": "4DPRH8TCGXAYPZWD",
    "imported_id": "4DPRH8TCGXAYPZWD",
    "imported_payee": "Maecenas PC",
    "notes": "Online-Einkauf ZKB Visa Debit Card Nr. xxxx 0193, Maecenas PC",
    "payee_name": "Maecenas PC",
  },
  {
    "account": "test-account-id",
    "amount": -1000,
    "cleared": true,
    "date": "2001-02-04",
    "id": "YEP66MP33MMUH",
    "imported_id": "YEP66MP33MMUH",
    "imported_payee": "Crusty Crab AG",
    "notes": "Gutschrift: Auftrags-Nr. xxx (Patty Reimbursement)",
    "payee_name": "Crusty Crab AG",
  },
  {
    "account": "test-account-id",
    "amount": 25850,
    "cleared": true,
    "date": "2001-02-04",
    "id": "C4CBEGNHGCABB",
    "imported_id": "C4CBEGNHGCABB",
    "imported_payee": "Tempus Scelerisque Incorporated",
    "notes": "Belastung eBanking Mobile: Auftrags-Nr. xxx (Montly Payment for Pineapple Insurance)",
    "payee_name": "Tempus Scelerisque Incorporated",
  },
  {
    "account": "test-account-id",
    "amount": -12925,
    "cleared": true,
    "date": "2001-02-04",
    "id": "PJ3HJVPJJBKRXBN6-2",
    "imported_id": "PJ3HJVPJJBKRXBN6-2",
    "imported_payee": " TWINT: Star, Patrick xxx",
    "notes": "Gutschrift TWINT: Star, Patrick xxx",
    "payee_name": " TWINT: Star, Patrick xxx",
  },
  {
    "account": "test-account-id",
    "amount": 500000,
    "cleared": true,
    "date": "2001-02-03",
    "id": "EWAYN6BUVBEPD",
    "imported_id": "EWAYN6BUVBEPD",
    "imported_payee": "Patrick Star",
    "notes": "Belastung eBanking Mobile: Auftrags-Nr. xxx",
    "payee_name": "Patrick Star",
  },
  {
    "account": "test-account-id",
    "amount": 139,
    "cleared": true,
    "date": "2001-02-03",
    "id": "NXUGJRVXUW8BE8EM",
    "imported_id": "NXUGJRVXUW8BE8EM",
    "imported_payee": "Nunc Mauris Limited",
    "notes": "Online-Einkauf aus Abonnement ZKB Visa Debit Card Nr. xxxx 0xxx, Nunc Mauris Limited",
    "payee_name": "Nunc Mauris Limited",
  },
  {
    "account": "test-account-id",
    "amount": 2185,
    "cleared": true,
    "date": "2001-02-03",
    "id": "SSZGZDY2QOMT8UZZ",
    "imported_id": "SSZGZDY2QOMT8UZZ",
    "imported_payee": "In Faucibus Inc.",
    "notes": "Einkauf ZKB Visa Debit Card Nr. xxxx 0xxx, In Faucibus Inc.",
    "payee_name": "In Faucibus Inc.",
  },
  {
    "account": "test-account-id",
    "amount": 390,
    "cleared": true,
    "date": "2023-02-01",
    "id": "TTKH7NNVIFGOEPG3-1",
    "imported_id": "TTKH7NNVIFGOEPG3-1",
    "imported_payee": "TWINT: Automat Mus Aenean Inc.",
    "notes": "Belastung TWINT: Automat Mus Aenean Inc.",
    "payee_name": "TWINT: Automat Mus Aenean Inc.",
  },
  {
    "account": "test-account-id",
    "amount": 380,
    "cleared": true,
    "date": "2023-02-01",
    "id": "JUGUSXJYLBPD9GT9-1",
    "imported_id": "JUGUSXJYLBPD9GT9-1",
    "imported_payee": "TWINT: Automat Mus Aenean Inc.",
    "notes": "Belastung TWINT: Automat Mus Aenean Inc.",
    "payee_name": "TWINT: Automat Mus Aenean Inc.",
  },
]
`;
